addon/components/frost-action-bar.js

```diff
@@ -67,38 +86,131 @@ export default Component.extend({
+  /**
+   * gets array of argument sets for building moreActionButtons button
+   * @param {array|object} controls - array or hash of controls
+   * @param {array} selectedItems - array of selected items for filtering controls
+   * @param {boolean|object|array} moreActionButtons - array or hash of extra controls
+   *  or flag indicating whether to generate them automaticaly
+   * @returns {array} array for building moreActionButtons button
+   */
+  _moreActionButtons (controls, selectedItems, moreActionButtons) {
+    controls = this.getNormalizedControls(controls, selectedItems)
+    moreActionButtons = this.getNormalizedControls(moreActionButtons, selectedItems)
+
+    // fail fast if we're told not use this feature
+    if (moreActionButtons === false) {
+      return []
+    }
+
+    // get the extra buttons if needed (default)
+    if (moreActionButtons === true) {
+      let buttonCount = controls.filter(control => control.name !== 'frost-button').length
+
+      // reverse to grab buttons from the end
+      moreActionButtons = controls.reverse()
+        .filter(control => control.name === 'frost-button' && ++buttonCount > MAX_CONTROLS)
+      moreActionButtons.reverse()
+    }
+
+    // convert buttons to POJOs if needed
+    return moreActionButtons.map(button => button.name === 'frost-button' ? this.convertButton(button) : button)
+  },
+
+  // == Functions =============================================================
+
+  /**
+   * converts a button to a hash of arguments to be used for building the moreActionButtons button
+   * @param {FrostButton} button - a frost-button to convert
+   * @returns {object} - plain object of props for moreActionButtons button
+   */
+  convertButton (button) {
+    const propNames = ['disabled', 'hook', 'onClick', 'text']
+
+    return button.args.named.keys.reduce((props, key) => {
+      // only get specified prop names
+      if (propNames.includes(key)) {
+        // compute the disabled property if exists and is needed
+        if (key === 'disabled') {
+          let disabled = get(button, 'args.named._map.disabled')
+          let compute = get(disabled, 'compute')
+
+          // need disabled.compute vs just compute for context
+          props[key] = typeof compute === 'function' ? disabled.compute() : disabled
+
+        // else grab value from component
+        } else {
+          props[key] = get(button, `args.named._map.${key}.inner`)
+        }
+      }
+
+      return props
+    }, {})
+  },
+
+  /**
+   * converts controls to array if they are given as a hash
+   * filters out inapplicable controls
+   * @param {object|array} controls - array or hash of controls
+   * @param {array} selectedItems - array of selected items for filtering
+   * @returns {array} - array of controls
+   */
+  getNormalizedControls (controls, selectedItems) {
+    // convert hash of controls to array if necessary
+    // notice this is Ember typeOf, which is very differnt from native typeof
```

Comment:
`differnt` -> `different`

Though don't think we need this comment.

Reply:
prob not, but it tripped me up first time I came to it.